extensions-core/kafka-extraction-namespace/src/main/java/io/druid/server/namespace/KafkaExtractionManager.java

```diff
@@ -0,0 +1,237 @@
+  private static final Logger log = new Logger(KafkaExtractionManager.class);
+
+  private final Properties kafkaProperties = new Properties();
+  private final ConcurrentMap<String, String> namespaceVersionMap = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, AtomicLong> topicEvents = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, ListenableFuture<?>> futures = new ConcurrentHashMap<>();
+  private final ListeningExecutorService executorService = MoreExecutors.listeningDecorator(
+      Executors.newCachedThreadPool(
+          new ThreadFactoryBuilder()
+              .setNameFormat("kafka-rename-consumer-%d")
+              .setDaemon(true)
+              .setPriority(Thread.MIN_PRIORITY)
+              .build()
+      )
+  );
+  private final AtomicInteger backgroundTaskCount = new AtomicInteger(0);
+
+  // Bindings in KafkaExtractionNamespaceModule
+  @Inject
+  public KafkaExtractionManager(
+      final Properties kafkaProperties
+  )
+  {
+    if (kafkaProperties.containsKey("group.id")) {
+      throw new IAE(
+          "Cannot set kafka property [group.id]. Property is randomly generated for you. Found [%s]",
+          kafkaProperties.getProperty("group.id")
+      );
+    }
+    if (kafkaProperties.containsKey("auto.offset.reset")) {
+      throw new IAE(
+          "Cannot set kafka property [auto.offset.reset]. Property will be forced to [smallest]. Found [%s]",
+          kafkaProperties.getProperty("auto.offset.reset")
+      );
+    }
+    this.kafkaProperties.putAll(kafkaProperties);
+    if (!this.kafkaProperties.containsKey("zookeeper.connect")) {
+      this.kafkaProperties.put("zookeeper.connect", "localhost:2181/kafka");
+    }
+    // Enable publish-subscribe
+    this.kafkaProperties.setProperty("auto.offset.reset", "smallest");
+  }
+
+  boolean supports(Properties otherProperties)
+  {
+    if (otherProperties.containsKey("zookeeper.connect")) {
+      return this.kafkaProperties.getProperty("zookeeper.connect")
+          .equals(otherProperties.getProperty("zookeeper.connect"));
+    }
+    return false;
+  }
+
+  public long getBackgroundTaskCount()
+  {
+    return backgroundTaskCount.get();
+  }
+
+  private static final Decoder<String> defaultStringDecoder = new Decoder<String>()
+  {
+    @Override
+    public String fromBytes(byte[] bytes)
+    {
+      return StringUtils.fromUtf8(bytes);
+    }
+  };
+
+  public long getNumEvents(String namespace)
+  {
+    if(namespace == null){
+      return 0L;
+    } else {
+      final AtomicLong eventCounter = topicEvents.get(namespace);
+      if(eventCounter != null) {
+        return eventCounter.get();
+      } else {
+        return 0L;
+      }
+    }
+  }
+
+  public void addListener(final String id, final KafkaExtractionNamespace kafkaNamespace, final Map<String, String> map)
+  {
+    final String topic = kafkaNamespace.getKafkaTopic();
+    final ListenableFuture<?> future = executorService.submit(
+        new Runnable()
```

Comment:
should we surround this with try catch ? otherwise any exception will stop the runnable